doc/user-guide.md

```diff
@@ -233,22 +233,30 @@ Once this is done, there are two ways to run the operator:
 
 ### 1. Run as a Deployment inside the cluster
 
-Build the memcached-operator image and push it to a registry:
+**Note**: `operator-sdk build` invokes `docker build`. Make sure your docker daemon is running and that you can run the docker client without sudo. You can check if this is the case by running `docker version`, which should complete without errors. Follow instructions for your OS/distribution on how to start the docker daemon and configure your access permissions, if needed.
```

Comment:
For the latest master, this isn't completely true. We now support using `buildah` as well. This note should mention that `docker` is the default, and that on some systems, `docker` doesn't work without root permissions by default, but `buildah` works without root.

Reply:
But in the example this user guide covers it's always docker, right? If that's the case, can we say

**Note**: The `operator-sdk build` command below invokes `docker build`

, to keep the note in the context of this user guide?

Reply:
We still need to explain the capabilities of `operator-sdk build`, since we're able to invoke either `buildah` or `docker` in that command as of now.

I recommend either leaving out that section entirely, since it doesn't directly relate to the issue this PR addresses, or add the following changes:

```md
**Note**: `operator-sdk build` invokes `docker build` by default, and optionally `buildah bud`. If using `buildah`, skip to the `operator-sdk build` invocation instructions below. If using `docker`, make sure your docker daemon is running and that you can run the docker client without sudo. You can check if this is the case by running `docker version`, which should complete without errors. Follow instructions for your OS/distribution on how to start the docker daemon and configure your access permissions, if needed.
```

Reply:
+1, changes included.